### PR TITLE
add script to lookup account by email address

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
# Purpose

PM and UR want to be able to help with customer queries and look up account details so they can do so.

Fixes UML-1188

## Approach

Scan, filter and get items.

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] The product team have tested these changes

### CODE REVIEW

File: scripts/analysis_scripts/lookup_account.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            region_name='eu-west-1',
+        )
+        session = sts.assume_role(
+            RoleArn=role_arn,
+            RoleSessionName='getting_service_statistics',
+            DurationSeconds=900
+        )
+        self.aws_iam_session = session
+
+    def get_viewer_codes(self):
```
Comment (GemTay) [inline scripts/analysis_scripts/lookup_account.py:56]: this function needs renaming

File: scripts/analysis_scripts/lookup_account.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        )
+        lpas = {}
+        for lpa in response['Items']:
+            lpas.update({lpa['SiriusUid']['S']:str(parser.isoparse(lpa['Added']['S']).date())})
+        return lpas
+
+    def write_csv(self,email_address):
+        print('Collecting data and writing CSV...')
+        n = 0
+        with open('some.csv', 'w', newline='') as f:
```
Comment (GemTay) [inline scripts/analysis_scripts/lookup_account.py:93]: the csv file could do with a better name

**Review by GemTay — CHANGES_REQUESTED** [review]
just some renaming needed
Also, did you test on a case where the LastLogin attribute doesnt exist?

**Comment by andrewpearce-digital** [comment]
> just some renaming needed
> Also, did you test on a case where the LastLogin attribute doesnt exist?

Thanks for the reminder. Added that in now. Will default to Never logged in if there's no last login date, and I've removed the filtering by login date as it's not necessary now that the output is updated.

File: scripts/analysis_scripts/lookup_account.py (hunk 1 above)
Comment (andrewpearce-digital) [reply scripts/analysis_scripts/lookup_account.py]: Fixed method and class naming

File: scripts/analysis_scripts/lookup_account.py (hunk 2 above)
Comment (andrewpearce-digital) [reply scripts/analysis_scripts/lookup_account.py]: fixed CSV file naming now to be descriptive and include the date it was generated.

File: scripts/analysis_scripts/lookup_account.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    def write_csv(self,email_address):
+        print('Collecting data and writing CSV...')
+        count = 0
+        csv_filename = "account-lookup-{}.csv".format((date.today()))
+
+        with open(csv_filename, 'w', newline='') as csv_file:
+            writer = csv.writer(
+                csv_file, quoting=csv.QUOTE_NONNUMERIC)
+            writer.writerow(["email", "last_login_datetime","lpas_added"])
+            viewer_codes = self.get_actor_users()
```
Comment (GemTay) [inline scripts/analysis_scripts/lookup_account.py:79]: rename this variable too, I missed that before, sorry!

**Review by GemTay — CHANGES_REQUESTED** [review]
spotted one more thing that needs renaming (sorry!), everything else looks great :)

**Review by GemTay — APPROVED** [review]
nice job!

**Pull request merged**